rasterio/_base.pyx

```diff
@@ -809,13 +809,78 @@ cdef class DatasetBase(object):
         num_items = CSLCount(metadata)
         return dict(metadata[i].split('=', 1) for i in range(num_items))
 
-    def colorinterp(self, bidx):
-        """Returns the color interpretation for a band or None."""
-        cdef GDALRasterBandH band = NULL
+    property colorinterp:
 
-        band = self.band(bidx)
-        value = GDALGetRasterColorInterpretation(band)
-        return ColorInterp(value)
+        """Returns a sequence of ``ColorInterp.<enum>`` representing
+        color interpretation in band order.
+
+        Pattern for converting to a band mapping:
+
+            from collections import OrderedDict
+            with rasterio.open('image.tif') as src:
+                band_mapping = OrderedDict(
+                    zip(src.colorinterp, self.indexes))
+                band_mapping[1] == src.colorinterp[0]
+
+        A normal dictionary would also work, however an ``OrderedDict()``
+        makes setting color interpretation easier as order is significant.
+
+            from collections import OrderedDict
+            with rasterio.open('image.tif', 'r+') as src:
```

Comment:
@geowurster I'm not sure about this example - are you envisioning use cases where the user would normally be working with this as a `dict`?

Seems like for example purposes, this could be simplified down to:
```
src.colorinterp = [ColorInterp.blue, ColorInterp.green, ColorInterp.red]
```

Reply:
You're right.  I tend to use as a band mapping, so this example is more opinionated than it needs to be.